types/node/index.d.ts

```diff
@@ -1283,6 +1283,137 @@ declare module "http" {
         setTimeout(timeout: number, callback?: () => void): this;
         setNoDelay(noDelay?: boolean): void;
         setSocketKeepAlive(enable?: boolean, initialDelay?: number): void;
+
+        /**
+         * Event emitter
+         * The defined events on documents including:
+         * 1. abort
+         * 2. connect
+         * 3. continue
+         * 4. information
+         * 5. response
+         * 6. socket
+         * 7. timeout
+         * 8. upgrade
+         * 9. close
+         * 10. drain
+         * 11. error
+         * 12. finish
+         * 13. pipe
+         * 14. unpipe
+         */
+
+        addListener(event: "abort", listener: () => void): this;
+        addListener(event: "connect", listener: (response: IncomingMessage, socket: net.Socket, head: Buffer) => void): this;
+        addListener(event: "continue", listener: () => void): this;
+        addListener(event: "information", listener: (information: { statusCode: number }) => void): this;
+        addListener(event: "response", listener: (response: IncomingMessage) => void): this;
+        addListener(event: "socket", listener: (socket: net.Socket) => void): this;
+        addListener(event: "timeout", listener: () => void): this;
+        addListener(event: "upgrade", listener: (response: IncomingMessage, socket: net.Socket, head: Buffer) => void): this;
+        addListener(event: "close", listener: () => void): this;
```

Comment:
`close, drain, error, finish, unpipe` are already defined in the base class, remove these please.

Reply:
I didn't have them originally, then realized that typescript won't read the base classes. 

The `addListener(event: string | symbol, listener: (...args: any[]) => void): this;` will override any base class. If I don't include that line, then there's a property/signature mismatch.

The same is done with `Http2Stream`.